pyccel/ast/core.py

```diff
@@ -2150,6 +2151,60 @@ def prefix(self):
         """
         return self._prefix
 
+class KernelCall(FunctionCall):
+    """
+    Represents a kernel call
+    (e.g. a[c, b]())
+
+
+
+    Parameters
+    ==========
+    func             : FunctionDef
+                       The definition of the function being called
+    args             : tuple
+                       The arguments being passed to the function
+
+    numBlocks        : NativeInteger
+                       The number of blocks
+    tpblock          : NativeInteger
+                       The number of threads per block
+    """
+    __slots__ = ('_numBlocks','_tpblock', '_func', '_args')
+    _attribute_nodes = (*FunctionCall._attribute_nodes, '_numBlocks', '_tpblock')
+
+    def __init__(self, func, args, numBlocks, tpblock, current_function=None):
+
+        self._func = func
+        self._args = args
+        self._numBlocks = numBlocks
+        self._tpblock = tpblock
+        super().__init__(func, args, current_function)
+
+    @property
+    def func(self):
+        """ The number of blocks in which the kernel will run
+        """
+        return self._func
```

Comment:
Why is this property needed as well as the inherited `funcdef` property?

Reply:
fixed in [0a8272c](https://github.com/pyccel/pyccel-cuda/pull/12/commits/0a8272c4f666d4d4b5a2b920daf7bb440400e205)